src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -689,6 +709,142 @@ MarkNodeHasMetadata(char *nodeName, int32 nodePort, bool hasMetadata)
 
 
 /*
+ * SequenceDDLCommandsForTable returns a list of commands which create sequences (and
+ * their schemas) to run on workers before creating the relation. The sequence creation
+ * commands are wrapped with a `worker_apply_sequence_command` call, which sets the
+ * sequence space uniquely for each worker. Notice that this function is relevant only
+ * during metadata propagation to workers and adds nothing to the list of sequence
+ * commands if none of the workers is marked as receiving metadata changes.
+ */
+List *
+SequenceDDLCommandsForTable(Oid relationId)
+{
+	List *sequenceDDLList = NIL;
+	List *ownedSequences = getOwnedSequences(relationId);
+	ListCell *listCell;
+	char *ownerName = TableOwner(relationId);
+	bool hasMetadataWorkers = HasMetadataWorkers();
+
+	if (!hasMetadataWorkers)
+	{
+		return sequenceDDLList;
```

Comment:
Moved the `hasMetadataWorkers` check inside `EnsureSupportedSequenceColumnType` function